pkg/bot/mattermost.go

```diff
@@ -213,18 +217,27 @@ func (b MMBot) getUser() *model.User {
 }
 
 // Create channel if not present and add botkube user in channel
-func (b MMBot) getChannel() *model.Channel {
+func (b MMBot) getChannelInfo(channelName string) *model.Channel {
 	// Checking if channel exists
-	botChannel, resp := b.APIClient.GetChannelByName(b.ChannelName, b.getTeam().Id, "")
+	botChannel, resp := b.APIClient.GetChannelByName(channelName, b.getTeam().Id, "")
 	if resp.Error != nil {
-		log.Fatalf("There was a problem finding Mattermost channel %s. %s", b.ChannelName, resp.Error)
+		log.Fatalf("There was a problem finding Mattermost channel %s. %s", channelName, resp.Error)
```

Comment:
Let's not use Fatal here and in getChannelById. We should return and log at the caller side. This can be addressed as a separate issue as well. Please file an issue if want to take it as separate task